### PR TITLE
Allow nested params when using Pages.GroupByParam and Pages.GroupByParamDate

### LINKED ISSUE #12289: Support nested params when using Pages.GroupByParam
### What version of Hugo are you using (`hugo version`)?
<pre>
$ hugo version
hugo v0.113.0+extended linux/amd64 BuildDate=2024-01-23T04:15:35Z VendorInfo=ubuntu:0.113.0-3ubuntu0.1
</pre>

### Does this issue reproduce with the latest release?
I am not sure

### Description
When using Pages.GroupByParam hugo will fail to build if the param used is a nested like the following example.

Frontmatter:
<pre>
{
   "author": {
       "email": "jsmith@example.org",
       "name": "John Smith"
   },
}
</pre>

Grouping fails
<pre>
{{ range .Pages.GroupByParam "author.name" }}
</pre>

But sorting works
<pre>
{{ range .Pages.ByParam "author.name" }}
</pre>

### Error
<pre>
execute of template failed at <.Pages.GroupByParam>: error calling GroupByParam: there is no such a param
</pre>

I reported this issue on discourse.gohugo.io and was asked to submit this ticket.
https://discourse.gohugo.io/t/how-to-use-groupbyparam-with-param-subkeys/48929

### PR DESCRIPTION
**Pull request by n1xx1**
Basically just use `maps.GetNestedParam` in `resource.GetParam` and `resource.GetParamToLower`. It only affects the `Pages.GroupByParam` and `Pages.GroupByParamDate` methods since it's only used there. There is no need to check for an error because `maps.GetNestedParam` never returns one. The tests are the same as the not nested version but using a nested field.

Fixes #12289.

### CODE REVIEW

File: resources/page/pagegroup_test.go [hunk 1]
```diff
@@ -372,6 +396,24 @@ func TestGroupByParamDate(t *testing.T) {
...
+	t.Parallel()
+	pages := preparePageGroupTestPages(t)
+	expect := PagesGroup{
+		{Key: "2012-04", Pages: Pages{pages[4], pages[2], pages[0]}},
+		{Key: "2012-03", Pages: Pages{pages[3]}},
+		{Key: "2012-01", Pages: Pages{pages[1]}},
+	}
+
+	groups, err := pages.GroupByParamDate("custom_object.date", "2006-01")
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (n1xx1) [reply resources/page/pagegroup_test.go]: I've never used quicktest or go-cmp, but I think it's correct. Let me know if it's ok or you need something else instead.

File: resources/page/pagegroup_test.go [hunk 2]
```diff
@@ -75,15 +81,13 @@ func TestGroupByWithFieldNameArg(t *testing.T) {
...
 
 	groups, err := pages.GroupBy(context.Background(), "Weight")
-	if err != nil {
-		t.Fatalf("Unable to make PagesGroup array: %s", err)
-	}
-	if !reflect.DeepEqual(groups, expect) {
-		t.Errorf("PagesGroup has unexpected groups. It should be %#v, got %#v", expect, groups)
-	}
+	c.Assert(err, qt.IsNil)
+	c.Assert(groups, comparePageGroup, expect)
```
Comment (bep) [inline resources/page/pagegroup_test.go:85]: I suspect qt.DeepEquals should do the job here?

Comment (n1xx1) [reply resources/page/pagegroup_test.go:85]: It doesn't, go-cmp can't compare unexported fields, so you can't compare `Page`s (I tried before making a custom comparer 😅)

Comment (n1xx1) [reply resources/page/pagegroup_test.go:85]: I'm just now noticing that you can use `cmpopts.IgnoreUnexported` or `AllowUnexported`, oh well. I can fix it or leave it like this, let me know.

Comment (bep) [reply resources/page/pagegroup_test.go:85]: No, leave it, it makes sense ...

**Pull request merged**